Wikipedia/assets/codemirror/codemirror-editTextSelection.js

```diff
@@ -6,92 +6,59 @@ class SelectedAndAdjacentText {
-
-    // Keep only the last 'maxAdjacentWordsToUse' words of 'textBeforeSelectedText'
-    const shouldKeepWordBeforeSelection = (e, i, a) => (a.length - i - 1) < maxAdjacentWordsToUse
-    // Keep only the first 'maxAdjacentWordsToUse' words of 'textAfterSelectedText'
-    const shouldKeepWordAfterSelection = (e, i) => i < maxAdjacentWordsToUse
-    const wordsBefore = this.textBeforeSelectedText.split(' ').filter(shouldKeepWordBeforeSelection).join(' ')
-    const wordsAfter = this.textAfterSelectedText.split(' ').filter(shouldKeepWordAfterSelection).join(' ')
-
-    const selectedTextPattern = replaceSpaceWith(this.selectedText, patternForSpace)
-    const textBeforeSelectedTextPattern = replaceSpaceWith(wordsBefore, patternForSpace)
-    const textAfterSelectedTextPattern = replaceSpaceWith(wordsAfter, patternForSpace)
-
-    // Attempt to locate wikitext selection based on the non-wikitext context strings above.
-    const atLeastOneNonWordOrOptionalStringPattern = '(?:\\W+|.*)'
-    const pattern = `(${textBeforeSelectedTextPattern.length > 0 ? '.*?' : ''}${textBeforeSelectedTextPattern}${atLeastOneNonWordOrOptionalStringPattern})(${selectedTextPattern})${atLeastOneNonWordOrOptionalStringPattern}${textAfterSelectedTextPattern}`
-    const regex = new RegExp(pattern, 's')
-
-    return regex
-  }
+const wordsOnly = string => string.replace(/\(.*?\)/g, '').replace(/{{.*}}/g, '').replace(/\W+/g, ' ').trim()
+const replaceSpaceWith = string => string.replace(/\s+/g, '\\W*(?:(?:[<{][^>]*[}>])*)\\W*')
+
+const adjacentComparisonWordCount = 6
+const adjacentCharsToGather = 200
+
+const scoreForMatch = (match, lastIndex, wordsBeforeFromHTML, wordsAfterFromHTML) => {
+  const wordsBeforeFromWikitext = getNumberOfWordsFromEndOfString(wordsOnly(match.input.substring(Math.max(0, match.index - adjacentCharsToGather), match.index)), adjacentComparisonWordCount)
+  const wordsAfterFromWikitext = getNumberOfWordsFromBeginningOfString(wordsOnly(match.input.substring(lastIndex, lastIndex + adjacentCharsToGather)), adjacentComparisonWordCount)
+  const wordsAfterScore = calculateScore(wordsAfterFromHTML, wordsAfterFromWikitext)
+  const wordsBeforeScore = calculateScore(wordsBeforeFromHTML.slice().reverse(), wordsBeforeFromWikitext.slice().reverse())
+  return wordsBeforeScore + wordsAfterScore
 }
 
 const wikitextRangeForSelectedAndAdjacentText = (selectedAndAdjacentText, wikitext) => {
-  const regex = selectedAndAdjacentText.regexForLocatingSelectedTextInWikitext(wikitext)
-  if (regex === null) {
-    return null
+  const wordsBeforeFromHTML = getNumberOfWordsFromEndOfString(selectedAndAdjacentText.textBeforeSelectedText, adjacentComparisonWordCount)
+  const wordsAfterFromHTML = getNumberOfWordsFromBeginningOfString(selectedAndAdjacentText.textAfterSelectedText, adjacentComparisonWordCount)
+  const selectedTextRegexPattern = `(${selectedAndAdjacentText.selectedText.replace(/\s+/g, '\\W*(?:(?:[<{][^>]*[}>])*)\\W*')})`
```

Comment:
Could the in-between-words regex be updated to be more permissive of text within links? I had some luck with `((\\[\\[[^\\]\\|]+\\|)|{{[^}]*}}|<[^>]*>|\\W)+` but unsure if that has any other consequences. The case I was looking at was selecting "is a grammatically correct sentence in American English" in [this article](https://en.wikipedia.org/wiki/Buffalo_buffalo_Buffalo_buffalo_buffalo_buffalo_Buffalo_buffalo)

Reply:
ah good catch! will give it a spin!

Reply:
works great! fixed!